api/v1alpha1/mirrorpeer_webhook.go

```diff
@@ -0,0 +1,135 @@
+)
+
+// log is for logging in this package.
+var mirrorpeerlog = logf.Log.WithName("mirrorpeer-webhook")
+
+const (
+	WebhookCertDir  = "/apiserver.local.config/certificates"
+	WebhookCertName = "apiserver.crt"
+	WebhookKeyName  = "apiserver.key"
+)
+
+func (r *MirrorPeer) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	bldr := ctrl.NewWebhookManagedBy(mgr).
+		For(r)
+
+	srv := mgr.GetWebhookServer()
+	srv.CertDir = WebhookCertDir
+	srv.CertName = WebhookCertName
+	srv.KeyName = WebhookKeyName
+
+	return bldr.Complete()
+}
+
+//+kubebuilder:webhook:path=/mutate-multicluster-odf-openshift-io-v1alpha1-mirrorpeer,mutating=true,failurePolicy=fail,sideEffects=None,groups=multicluster.odf.openshift.io,resources=mirrorpeers,verbs=create;update,versions=v1alpha1,name=mmirrorpeer.kb.io,admissionReviewVersions=v1;v1beta1
+
+var _ webhook.Defaulter = &MirrorPeer{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *MirrorPeer) Default() {
+	if len(r.Spec.SchedulingIntervals) == 0 {
+		mirrorpeerlog.Info("No values set for schedulingIntervals, defaulting to 5m", "MirrorPeer", r.Name)
+		r.Spec.SchedulingIntervals = []string{"5m"}
+	}
+}
+
+//+kubebuilder:webhook:path=/validate-multicluster-odf-openshift-io-v1alpha1-mirrorpeer,mutating=false,failurePolicy=fail,sideEffects=None,groups=multicluster.odf.openshift.io,resources=mirrorpeers,verbs=create;update;delete,versions=v1alpha1,name=vmirrorpeer.kb.io,admissionReviewVersions=v1;v1beta1
+
+var _ webhook.Validator = &MirrorPeer{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *MirrorPeer) ValidateCreate() error {
+	mirrorpeerlog.Info("validate create", "name", r.Name)
+
+	return r.ValidateMirrorPeer(r)
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *MirrorPeer) ValidateUpdate(old runtime.Object) error {
+	mirrorpeerlog.Info("validate update", "name", r.Name)
+	oldMirrorPeer, ok := old.(*MirrorPeer)
+	if !ok {
+		return fmt.Errorf("error casting old object to MirrorPeer")
+	}
+
+	if len(r.Spec.Items) != len(oldMirrorPeer.Spec.Items) {
+		return fmt.Errorf("error updating MirrorPeer, new and old spec.items have different lengths")
+	}
+
+	refs := make(map[string]int)
+	for idx, pr := range oldMirrorPeer.Spec.Items {
+		// Creating a make-shift set of references to check for duplicates
+		refs[fmt.Sprintf("%s-%s-%s", pr.ClusterName, pr.StorageClusterRef.Namespace, pr.StorageClusterRef.Name)] = idx
+	}
+
+	for _, pr := range r.Spec.Items {
+		key := fmt.Sprintf("%s-%s-%s", pr.ClusterName, pr.StorageClusterRef.Namespace, pr.StorageClusterRef.Name)
+		if _, ok := refs[key]; !ok {
+			return fmt.Errorf("error validating update: new MirrorPeer %s references a StorageCluster %s/%s that is not in the old MirrorPeer", r.Name, pr.StorageClusterRef.Namespace, pr.StorageClusterRef.Name)
+		}
+	}
+	return r.ValidateMirrorPeer(r)
```

Comment:
```suggestion
	return r.ValidateMirrorPeer()
```
No need to pass `r` when you're already using `r.ValidateMirrorPeer`.

Reply:
Changed to validateMirrorPeer(r)